src/ui/marker.js

```diff
@@ -11,26 +11,53 @@ import type Popup from './popup';
 import type {LngLatLike} from "../geo/lng_lat";
 import type {MapMouseEvent} from './events';
 
+export type Anchor = 'middle' | 'top' | 'bottom' | 'left' | 'right' | 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
+export type Offset = number | PointLike | {[Anchor]: PointLike};
+
+export type MarkerOptions = {
+    anchor: Anchor,
+    offset: Offset
+}
+
 /**
  * Creates a marker component
- * @param element DOM element to use as a marker. If left unspecified a default SVG will be created as the DOM element to use.
- * @param options
- * @param options.offset The offset in pixels as a {@link PointLike} object to apply relative to the element's center. Negatives indicate left and up.
+ * @param {Object} [element] DOM element to use as a marker. If left unspecified a default SVG will be created as the DOM element to use.
+ * @param {Object} [options]
+ * @param {Array<string> | string} [options.anchor = Anchor] - A string indicating the markers's location relative to
```

Comment:
See "An optional parameter and default value" at http://usejsdoc.org/tags-param.html the default value should be `middle`.

See https://github.com/mapbox/mapbox-gl-js/blob/master/src/ui/popup.js#L37, the options.anchor type is only `string`. Each marker can only have one anchor so the type should just be string.